src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpIsAndCastCheckTests.cs

```diff
@@ -0,0 +1,456 @@
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if ((x ? y : z) is string v)
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestInlineTypeCheckWithElse()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if (x is string)
+        {
+            [|var|] v = (string)x;
+        } 
+        else
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (x is string v)
+        {
+        } 
+        else
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if (x is string)
+        {
+            // prefix comment
+            [|var|] v = (string)x;
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        if (x is string v)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if (x is string)
+        {
+            [|var|] v = (string)x; // suffix comment
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // suffix comment
+        if (x is string v)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestComments3()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if (x is string)
+        {
+            // prefix comment
+            [|var|] v = (string)x; // suffix comment
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        // prefix comment
+        // suffix comment
+        if (x is string v)
+        {
+        }
+    }
+}", compareTokens: false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheckParenthesized1()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if ((x) is string)
+        {
+            [|var|] v = (string)x;
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if ((x) is string v)
+        {
+        } 
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheckParenthesized2()
+        {
+            await TestAsync(
+@"class C
+{
+    void M()
+    {
+        if (x is string)
+        {
+            [|var|] v = (string)(x);
+        } 
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        if (x is string v)
+        {
+        } 
+    }
+}");
```

Comment:
Never mind. I see how you did it now. When the edge cases come in, we can revisit what simplifications would be needed.